File: 3D Multiplayer GP2/Assets/Modern Day Characters/Base Male/Scripts/C#/GUI/ButtonPosition.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

[RequireComponent(typeof(GUITexture))]

public class ButtonPosition : MonoBehaviour {
	public bool yPositionFromTop = true;
	public bool xPositionFromRight = false;
	public Vector2 buttonPosition;
	public float buttonSize = 1.0f;
	public Rect getGuiTexture;

	void Start() {
		getGuiTexture = GetComponent<GUITexture>().pixelInset;
		transform.position = Vector3.zero;
	}

	void Update() {
		if(xPositionFromRight) {
			getGuiTexture.x = Screen.width - buttonPosition.x - GetComponent<GUITexture>().texture.width * 0.5f * buttonSize;
		} else {
			getGuiTexture.x = buttonPosition.x - GetComponent<GUITexture>().texture.width * 0.5f * buttonSize;
		}

		if(yPositionFromTop) {
			getGuiTexture.y = Screen.height - buttonPosition.y - GetComponent<GUITexture>().texture.height * 0.5f * buttonSize;
		} else {
			getGuiTexture.y = buttonPosition.y - GetComponent<GUITexture>().texture.height * 0.5f * buttonSize;
		}

		getGuiTexture.width = GetComponent<GUITexture>().texture.width * buttonSize;
		getGuiTexture.height = GetComponent<GUITexture>().texture.height * buttonSize;
	}
}

File: 3D Multiplayer GP2/Assets/Modern Day Characters/Base Male/Scripts/JavaScript/GUI/ButtonPosition.js
#pragma strict
@script RequireComponent(GUITexture)

public var yPositionFromTop : boolean = true;
public var xPositionFromRight : boolean = false;
public var buttonPosition : Vector2;
public var buttonSize : float = 1.0;
public var getGuiTexture : GUITexture;

function Start() {
	getGuiTexture = GetComponent(GUITexture);
	transform.position = Vector3.zero;
}

function Update() {
	if(xPositionFromRight) {
		GetComponent.<GUITexture>().pixelInset.x = Screen.width - buttonPosition.x - GetComponent.<GUITexture>().texture.width * 0.5 * buttonSize;
	} else {
		GetComponent.<GUITexture>().pixelInset.x = buttonPosition.x - GetComponent.<GUITexture>().texture.width * .5 * buttonSize;
	}

	if(yPositionFromTop) {
		GetComponent.<GUITexture>().pixelInset.y = Screen.height - buttonPosition.y - GetComponent.<GUITexture>().texture.height * 0.5 * buttonSize;
	} else {
		GetComponent.<GUITexture>().pixelInset.y = buttonPosition.y - GetComponent.<GUITexture>().texture.height * 0.5 * buttonSize;
	}

	GetComponent.<GUITexture>().pixelInset.width = GetComponent.<GUITexture>().texture.width * buttonSize;
	GetComponent.<GUITexture>().pixelInset.height = GetComponent.<GUITexture>().texture.height * buttonSize;
}